app/views/results/student/_student_summary.html.erb

```diff
@@ -85,9 +93,18 @@
         <span><%= result.get_subtotal %> / <%= assignment.max_mark %></span>
     </div>
     <% end %>
-    <% if assignment.submission_rule.is_a? GracePeriodSubmissionRule %>
-      <%= render partial: assignment.submission_rule.grader_tab_partial,
-                 locals: { grouping: result.submission.grouping } %>
+      <% if assignment.submission_rule.is_a? GracePeriodSubmissionRule and
+          (result.submission.grouping.grace_period_deductions.
```

Comment:
You can just do `result.submission.grouping.grace_period_deductions.any?` to be a bit cleaner. The `includes` is only necessary down below.